frontend/src/hooks/useAsyncData.ts

```diff
@@ -9,31 +9,87 @@ import {
 import useEvent from "react-use-event-hook";
 import { invariant } from "@/utils/invariant";
 
-interface PendingResult {
+/**
+ * Base result interface containing common properties for all async data states.
+ *
+ * @template T - The type of data being fetched
+ */
+export interface AsyncBaseResult<T> {
```

Comment:
This is ideal for documentation, but I think we should avoid exporting and encourage the use of only `AsyncDataResult<T>` outside of this module.

```suggestion
interface AsyncBaseResult<T> {
```